recipe-client-addon/lib/AboutStudiesProtocol.jsm

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu, interfaces: Ci, manager: Cm, results: Cr} = Components;
+Cm.QueryInterface(Ci.nsIComponentRegistrar);
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+this.EXPORTED_SYMBOLS = ["AboutStudiesProtocol"];
+
+/**
+ * Component definition for the about:studies protocol handler.
+ * Registers a component with the browser that establishes an `about:studies`
+ * protocol handler. Navigating to `about:studies` displays the
+ * `AboutStudies.xml` file.
+ */
+function StudiesProtocolHandler() {}
+StudiesProtocolHandler.prototype = {
```

Comment:
I wanted to keep the `uri`, `classID`, etc tied to the ProtocolHandler class, so I just defined everything in one go. Using es6 classes we have to do something like:
```
class Whatever { ... }
Whatever.prototype = { thing: 'woo' }
```

Alternatively, we could define the `uri` etc in the class constructor, however I suspect the `Cm.registerFactory` call looks up the handler info from the prototype. I will investigate and see what we can do.